### DISCUSSION TITLE
Need numeric samples instead of String when exported to mat file.

### DISCUSSION THREAD

**Discussion by axaiem** [danielhrisca/asammdf#679] [Q&A]
I have a signal for example SteerigStatus which can have values either "active" or "inactive". When  I export this signal to .mat I need the values to be 0 or 1 not "active" or "inactive". Is there a way to do this? (I am able to achieve this through asammdf gui version by selecting "Raw values" but unsuccessful programmatically) .

**Comment by david32** [comment]
Does it happen when calling export() with the argument/option ```raw= True```?

**Reply by dglent** [reply]
i use this option when i read a signal in case where i don't need the conversions

**Reply by axaiem** [reply]
For both arguments raw = True and raw =False. I get the same output i.e String values in sample.

**Reply by dglent** [reply]
Can you see what contains the "Conversions" attribute of the signal ?

**Comment by danielhrisca** [comment]
Call export with ``ignore_value2text_conversions=True``

**Reply by axaiem** [reply]
Tried this. Doesn't work! Could I be doing any mistake?

**Comment by danielhrisca** [comment]
right click the channel in the plot selection tree and share the information
![image](https://user-images.githubusercontent.com/20952040/161482064-4751e363-6ee3-4e0a-a297-305b93dbd638.png)
![image](https://user-images.githubusercontent.com/20952040/161482083-f5a0ceba-9258-445d-97a8-e70f643c31eb.png)

**Comment by david32** [comment]
@danielhrisca 

I don't know if you would count it as the same issue.
I'm running this code on a signal with a TABX conversion from int to byte.

```python
def testGetRawSignal():
    mdfFile = MDF(f'D:\\XXXX\\mdfTests\\TestSignal.mf4')

    orgSig = mdfFile.get("XXXXXState")
    print(f'{orgSig.samples[:10]}')

    newSig = orgSig.physical()
    print(f'{newSig.samples[:10]}')
```

The result is the same array printed, in both cases bytes (after conversion)
```python
[b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init'
 b'Init']
[b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init' b'Init'
 b'Init']
```

Is it the same or should I open a new issue?

**Reply by danielhrisca** [reply]
Call get with ``raw=True``

**Reply by david32** [reply]
yes, that is an option.
But it means that the first signal that I get would be raw. 
And Physical() should return the raw signal as far as I understand. corrent?

**Reply by danielhrisca** [reply]
Physical applies the conversion rule if the signal has one

**Reply by david32** [reply]
Yes, 
```
**get the physical samples values**
```
I thought physical would mean raw.

So physical() is to be used when get() was called with `raw=True` when we want the conversion applied. 
Perhaps its possible to also make a function to do the opposite? return a new signal object with raw values instead of the converted ones?

[I'm iterating over all channels in the MDF file (making a dictionary at the start of the script for future performance) and they return after conversion, so sometimes I want to get the raw version of a certain signal]
